src/main/java/org/jitsi/jigasi/transcription/GoogleCloudTranscriptionService.java

```diff
@@ -0,0 +1,642 @@
+                    .build();
+
+            RecognizeResponse recognizeResponse =
+                    client.recognize(config, audio);
+
+            client.close();
+
+            StringBuilder builder = new StringBuilder();
+            for (SpeechRecognitionResult result :
+                    recognizeResponse.getResultsList())
+            {
+                builder.append(result.toString());
+                builder.append(" ");
+            }
+
+            String transcription = builder.toString().trim();
+            resultConsumer.accept(new TranscriptionResult(transcription));
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public StreamingRecognitionSession initStreamingSession()
+        throws UnsupportedOperationException
+    {
+        return new GoogleCloudStreamingRecognitionSession();
+    }
+
+    @Override
+    public boolean supportsFragmentTranscription()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean supportsStreamRecognition()
+    {
+        return true;
+    }
+
+    /**
+     * A Transcription session for transcribing streams for Google Cloud
+     * speech-to-text API
+     */
+    public class GoogleCloudStreamingRecognitionSession
+        implements StreamingRecognitionSession
+    {
+
+        /**
+         * The SpeechClient which will be used to initiate and to end the
+         * session
+         */
+        private SpeechClient client;
+
+        /**
+         * The ApiStreamObserver which will retrieve new incoming transcription
+         * results
+         */
+        private ResponseApiStreamingObserver<StreamingRecognizeResponse>
+            responseObserver = new ResponseApiStreamingObserver<>();
+
+        /**
+         * A manager which acts as a ApiStreamObserver which will send new audio
+         * request to be transcribed
+         */
+        private RequestApiStreamObserverManager requestManager;
+
+        /**
+         * A single thread which is used to sent all requests to the API.
+         * This is needed to reliably sent the first request to the service
+         */
+        private ExecutorService service = Executors.newSingleThreadExecutor();
+
+        /**
+         * Create a new session with the Google Cloud API
+         */
+        private GoogleCloudStreamingRecognitionSession()
+        {
+            try
+            {
+                this.client = SpeechClient.create();
+                this.requestManager
+                    = new RequestApiStreamObserverManager(client,
+                                                          responseObserver);
+            }
+            catch(Exception e)
+            {
+                e.printStackTrace();
+            }
+        }
+
+        @Override
+        public void give(final TranscriptionRequest request)
+        {
+            this.service.submit(() -> requestManager.sentRequest(request));
```

Comment:
I'm not sure sendRequest is a right name either, because we do not necessarily immediately send a request, but I've changed it